### PR TITLE
Parts and references

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?

No parts and references compartment in Block diagram

Issue Number: #138 

### What is the new behavior?

Parts and references are visible:

image

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

NB. It looks like SysML elements are not visible in the tree view at all (!)

### CODE REVIEW

File: gaphor/SysML/blocks/block.py [hunk 1]
```diff
@@ -1,9 +1,154 @@
...
+                and [operations_compartment(self.subject)]
+                or []
+            ),
+            *(
+                self.show_parts
+                and self.subject
+                and [
+                    self.block_compartment(
+                        gettext("parts"),
+                        lambda a: a.association and a.aggregation != "none",
```
Comment (danyeaw) [inline gaphor/SysML/blocks/block.py:108]: I think this should be `a.aggregation == "composite"`

File: gaphor/SysML/blocks/block.py [hunk 2]
```diff
@@ -1,9 +1,154 @@
...
+                ]
+                or []
+            ),
+            *(
+                self.show_references
+                and self.subject
+                and [
+                    self.block_compartment(
+                        gettext("references"),
+                        lambda a: a.association and a.aggregation == "none",
```
Comment (danyeaw) [inline gaphor/SysML/blocks/block.py:119]: I think this should be `a.aggregation == "none" or a.aggregation == "composite"`

**Comment by amolenaar** [comment]
I was a bit in doubt on the shared association. I'll change it accordingly.

File: gaphor/SysML/blocks/block.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/SysML/blocks/block.py]: Fixed

File: gaphor/SysML/blocks/block.py (hunk 2 above)
Comment (amolenaar) [reply gaphor/SysML/blocks/block.py]: And fixed :smile:

**Comment by amolenaar** [comment]
Removing attributes and operations is trivial now, since Block is a separate diagram item now, not inheriting from ClassItem.

Isn't it custom to have attributes for, for example, `maxPSI` on an `InflationValve` and a `thresholdPSI` on a `TirePressureMonitor`?

(using American units here, we use Bar as unit for pressure :smiley: )

**Comment by danyeaw** [comment]
Those would be added by a 3rd Property that we haven't created yet called a Value Property. To support those we would need to add some default ValueTypes and make it possible to add additional ones, since every value property is typed by a ValueType.

**Comment by amolenaar** [comment]
Then we'd better remove attributes and operations, before they are misused. Or maybe keep attributes until we have ValueTypes in place?

**Comment by danyeaw** [comment]
I would be ok with just removing them I think, and then once we get initial SysML support finished, we can come back in with additions like ValueTypes.

**Comment by amolenaar** [comment]
Fair enough. Sounds like a simple change to me.

**Comment by amolenaar** [comment]
I removed attributes and operations.

**Pull request merged**